Add functionality for reading sensewear xlsx files as externally derived epoch data

vincentvanhees opened the pull request:
<!-- Describe your PR here -->

Fixes #974 

- As you will see I am loading the METs column from the xlsx file as our main 'metric' for PA analysis, because we can use actual MET levels as thresholds to define LIPA and MVPA thresholds. However, I store MET under the metric name "ExtAct" (external activity), because I think it is not practical to have a new metric name for every external epoch datatype. GGIR would then become too much cluttered with terminology specific to uncommon data types. By storing it as "ExtAct", we can use "ExtAct" for all movement indicators from external software to be derived in the future if they do not overlap with existing metric names.
- Similarly I use "wake_sleep" as column name in the part 1 milestone data to make it align with previous work on utilising externally derived sleep classifications (project with Kalaivani and her Python based sleep classifier). So, also there I tried to make it generic and not Sensewear specific.

<!-- Please, make sure the following items are checked -->
Checklist before merging:

- [x] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [x] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [x] Updated or expanded the documentation.
- [x] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` file, if you think you made a significant contribution.

jhmigueles reviewed (commented):
Thanks Vincent! Only 2 minor suggestions:

1. I tried this with also an updated version of excel file (.xlsx instead of .xls) and it works well. Should the documentation mention that the code works with .xls and .xlsx? Just in case any user think they should reformat all their files before using GGIR.
2. In `g.part5_initialise_ts`, the code scales the external activity metric by 1000 as it would be a g unit metric. Line 5 of this functions should include ExtAct to avoid this behavior: `scale = ifelse(test = grepl("^Brond|^Neishabouri|^ZC|^ExtAct", params_general[["acc.metric"]]), yes = 1, no = 1000)`
3. I had to use argument `acc.metric = "ExtAct"` to avoid an error in part 5 when initialising the time series. I think this should be mentioned in the documentation. And this made me think whether we should give the option to the users to give a name to the external metric they might be using with an argument like: `ExtAct_name` which defaults to `"ExtAct"`.

jhmigueles commented on man/GGIR.Rd:
```diff
@@ -1612,6 +1613,21 @@ GGIR(mode = 1:5,
+           
+           
+       # For externally derived Sensear data in .xls format:  
+        GGIR(datadir = "C:/yoursenseweardatafolder",
```
Also, add argument `acc.metric = "ExtAct"` in the example, otherwise an error occurs in `g.part5_initialise_ts` as it would try to load ENMO.

vincentvanhees replied:
Thanks, thresholds now clarified in [91d1831](https://github.com/wadpac/GGIR/pull/1013/commits/91d1831f05e30d9895ef5c9173f642ed444720d6). Re. second point in [8ae4cc2](https://github.com/wadpac/GGIR/pull/1013/commits/8ae4cc2aa66841bbd005e1b7e0af0b08ef0c9042) I am now avoiding the need for `acc.metric = "ExtAct"` by autosetting it in check_params if data format is sensewear_xls, in that way it happens without the user needing to worry about it.

vincentvanhees commented on man/GGIR.Rd:
```diff
@@ -1612,6 +1613,21 @@ GGIR(mode = 1:5,
+            mode = 1:5,
+            windowsizes = c(60, 900, 3600),
+            dataFormat = "sensewear_xls",
+            extEpochData_timeformat = "\%d-\%b-\%Y \%H:\%M:\%S",
```
This is an Rd file not Rmd, so escape characters is necessary, user does not see it when documentation is compile.

vincentvanhees commented:
> I tried this with also an updated version of excel file (.xlsx instead of .xls) and it works well. Should the documentation mention that the code works with .xls and .xlsx? Just in case any user think they should reformat all their files before using GGIR.

Thanks, now addressed in [91d1831](https://github.com/wadpac/GGIR/pull/1013/commits/91d1831f05e30d9895ef5c9173f642ed444720d6)

> In g.part5_initialise_ts, the code scales the external activity metric by 1000 as it would be a g unit metric. Line 5 of this functions should include ExtAct to avoid this behavior: scale = ifelse(test = grepl("^Brond|^Neishabouri|^ZC|^ExtAct", params_general[["acc.metric"]]), yes = 1, no = 1000)

Thanks, now addressed in [b4c53ff](https://github.com/wadpac/GGIR/pull/1013/commits/b4c53ff1fd451229756db48b3796067340c24231)

> I had to use argument acc.metric = "ExtAct" to avoid an error in part 5 when initialising the time series. I think this should be mentioned in the documentation. 

I have now circumvented this by auto-setting acc.metric = "ExtAct" in check_params: [8ae4cc2](https://github.com/wadpac/GGIR/pull/1013/commits/8ae4cc2aa66841bbd005e1b7e0af0b08ef0c9042).

> And this made me think whether we should give the option to the users to give a name to the external metric they might be using with an argument like: ExtAct_name which defaults to "ExtAct".

I am not in favour of this, because:
- If we want the metric names to be more intuitive then it would be best to choose it ourselves inside GGIR rather than to let each individual user decide as that will complicate comparisons and interpretation of variable names. For example, when they come to use with an error and show us output with column names we are not familiar with or when they want to merge datasets across studies.
- If we want to facilitate more old sensor brands then there will be a growing number of output metric names, all of which then have to be accommodated in the GGIR code, increasing code complexity, while the beneficial value of working with these older and hardly used data types is limited. Note: Making the metric name more specific to each branch requires that part 1, 2 and 5 code need to be expanded.
- MET as metric name could give the impression that GGIR estimates METs, so we would need a name that reflects that not GGIR but Sensewear estimated the METs, but then that variable name would be longer. ExtAct has the advantage that we can re-use it for all exotic metrics from all external brands and acknowledge in the documentation what is being used from the sensor output for each of those brands.

jhmigueles reviewed (approved):
I agree with all the changes you've made and I don't have more comments, I think this is ready to be merged.
